### PR TITLE
docs: deprecate old security project

### LINKED ISSUE #204: Add more detailed deprecation information in feature docs and types for `Arcus.EventGrid.Security` project
**Is your feature request related to a problem? Please describe.**
The `Arcus.EventGrid.Security` project is being replaced by `Arcus.EventGrid.WebApi.Security` but is not really documented thoroughly to make the transition smoothly.

**Describe the solution you'd like**
Add more deprecation information in the feature docs and on the types of the deprecated project..

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add remaining obsolete types and add deprecation message on the feature docs page.

Closes #204

### CODE REVIEW

**Comment by stijnmoreels** [comment]
We already released the `Arcus.EventGrid.WebApi.Security` NuGet package, so we can already deprecate this other one, right? Gives consumers even more information.

File: docs/preview/02-Features/endpoint-validation.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ The features described here require the following package:
 PM> Install-Package Arcus.EventGrid.Security.WebApi
 ```
 
+> ⚠ This package used to be called `Arcus.EventGrid.Security`. Please make sure that you migrate towards this package as it's not being actively maintained beyond v3.2.
```
Comment (fgheysels) [inline docs/preview/02-Features/endpoint-validation.md:21]: Should we put this before the Install-Package command ?
It should definitely stand out that this is deprecated; could be put in bold ?

Comment (stijnmoreels) [reply docs/preview/02-Features/endpoint-validation.md:21]: We can do both 😉 .

Comment (stijnmoreels) [reply docs/preview/02-Features/endpoint-validation.md]: Done that.

**Review by fgheysels — APPROVED** [review]

**Comment by fgheysels** [comment]
> We already released the `Arcus.EventGrid.WebApi.Security` NuGet package, so we can already deprecate this other one, right? Gives consumers even more information.

Certainly.

I'd like to close this PR, but t seems there are some failing tests

**Comment by stijnmoreels** [comment]
I think something wrong with the Azure resources. `master` fails too. We'll see, maybe later 🤔

**Pull request merged**